src/lightning_utilities/core/imports.py

```diff
@@ -132,3 +135,41 @@ def get_dependency_min_version_spec(package_name: str, dependency_name: str) ->
         "This is an internal error. Please file a GitHub issue with the error message. Dependency "
         f"{dependency_name!r} not found in package {package_name!r}."
     )
+
+
+def requires(*module_path: str) -> Callable:
+    """Wrapper for early import failure with some nice exception message.
+
+    Example:
+
+        >>> @requires("libpath")
+        ... def my_cwd():
+        ...     from pathlib import Path
+        ...     return Path(__file__).parent
+
+        >>> class MyRndPower:
+        ...     @requires("math", "random")
+        ...     def __init__(self):
+        ...         from math import pow
+        ...         from random import randint
+        ...         self._rnd = pow(randint(1, 9), 2)
+
+    .. note:: For downgrading exception to warning you export `LIGHTING_TESTING=1` which is handu for testing
+    """
+
+    def decorator(func: Callable) -> Callable:
+        @functools.wraps(func)
+        def wrapper(*args: Any, **kwargs: Any) -> Any:
+            unavailable_modules = [module for module in module_path if not module_available(module)]
+            if any(unavailable_modules):
+                is_lit_testing = bool(int(os.getenv("LIGHTING_TESTING", "0")))
```

Comment:
This should use an envvar. Instead have a flag `raise=bool`

Reply:
Sorry, I meant shouldn't

Reply:
fixed